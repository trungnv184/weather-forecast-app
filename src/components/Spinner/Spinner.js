import React from "react";
import styles from "./Spinner.module.css";

const Spinner = () => <div className={styles.loader}>Loading data ...</div>;
export default Spinner;
