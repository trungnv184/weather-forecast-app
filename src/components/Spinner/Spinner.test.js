import React from "react";
import {render, cleanup} from "@testing-library/react";
import {screen} from "@testing-library/dom";
import Spinner from "./Spinner";

afterEach(cleanup);

test("should display loading data text", () => {
  render(<Spinner />);
  expect(screen.getByText(/Loading/i)).toHaveTextContent("Loading data ...");
});

test("snapshot", () => {
  const {container} = render(<Spinner />);
  expect(container.firstChild).toMatchSnapshot();
});
