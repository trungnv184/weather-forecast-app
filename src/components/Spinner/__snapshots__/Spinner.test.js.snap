// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="loader"
>
  Loading data ...
</div>
`;
